### PR TITLE
added swagger documentation to the existing endpoints

### PR DESCRIPTION
**Pull request by simionescuraluca**

### CODE REVIEW

File: src/main/java/com/vinyl/controller/ProductController.java [hunk 1]
```diff
@@ -3,20 +3,30 @@
...
 import com.vinyl.modelDTO.AddProductToCartDTO;
 import com.vinyl.service.ProductService;
+import io.swagger.annotations.*;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+@Api(value="Product Management", description="Operations pertaining to products in Product Management System")
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/ProductController.java]: please rename this to be more Vinyl store appropriate

File: src/main/java/com/vinyl/controller/UserController.java [hunk 2]
```diff
@@ -53,8 +72,15 @@
 		return new ResponseEntity<>(tokenDTO,HttpStatus.OK);
 	}
 
+	@ApiOperation(value = "User gets cart details", response = ResponseEntity.class)
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/UserController.java]: I think response is just a string :D

File: src/main/java/com/vinyl/controller/ProductController.java [hunk 3]
```diff
@@ -3,20 +3,30 @@
...
 @RestController
 public class ProductController {
 
     @Autowired
     ProductService productService;
 
+    @ApiOperation(value = "Add a product to the shopping cart", response = ResponseEntity.class)
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/ProductController.java]: also use @ApiModel on the DTOs

File: src/main/java/com/vinyl/controller/UserController.java [hunk 4]
```diff
@@ -15,14 +16,20 @@
...
+@Api(value="User Management", description="Operations pertaining to users in User Management System")
 @RestController
 public class UserController {
 
 	@Autowired
 	UserService userService;
 
+	@ApiOperation(value = "Create an account", response = ResponseEntity.class)
+	@ApiResponses(value = {
+			@ApiResponse(code = 201, message = "Successfully created user"),
```
Comment (dragosslujeru) [inline src/main/java/com/vinyl/controller/UserController.java]: see if you can extract duplicate string as constants and use them in these annotations

File: src/main/java/com/vinyl/controller/ProductController.java (hunk 1 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/ProductController.java]: done

File: src/main/java/com/vinyl/controller/UserController.java (hunk 2 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/UserController.java]: fixed

File: src/main/java/com/vinyl/controller/UserController.java (hunk 4 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/UserController.java]: done

File: src/main/java/com/vinyl/controller/ProductController.java (hunk 3 above)
Comment (simionescuraluca) [reply src/main/java/com/vinyl/controller/ProductController.java]: done

**Review by dragosslujeru — APPROVED** [review]

**Pull request merged**